test/integration/puppetlabs/services/master/plans_int_test.clj

```diff
@@ -0,0 +1,143 @@
+            [puppetlabs.puppetserver.testutils :as testutils]
+            [cheshire.core :as json]
+            [me.raynes.fs :as fs]))
+
+(def test-resources-dir
+  (ks/absolute-path "./dev-resources/puppetlabs/services/master/plans_int_test"))
+
+(defn plan-path
+  [plan-name]
+  (str test-resources-dir "/" plan-name))
+
+(defn purge-env-dir
+  []
+  (-> testutils/conf-dir
+      (fs/file "environments")
+      fs/delete-dir))
+
+(use-fixtures :once
+              (testutils/with-puppet-conf
+               (fs/file test-resources-dir "puppet.conf")))
+
+(use-fixtures :each
+              (fn [f]
+                (purge-env-dir)
+                (try
+                  (f)
+                  (finally
+                    (purge-env-dir)))))
+
+(def request-as-text-with-ssl (assoc testutils/ssl-request-options :as :text))
+
+(defn get-all-plans
+  [env-name]
+  (let [url (str "https://localhost:8140/puppet/v3/plans"
+                 (if env-name (str "?environment=" env-name)))]
+    (try
+      (http-client/get url request-as-text-with-ssl)
+      (catch Exception e
+        (throw (Exception. "plans http get failed" e))))))
+
+(defn get-plan-details
+  "plan-name is the plan's full name, e.g. 'apache::reboot'."
+  [env-name full-plan-name]
+  (let [[module-name plan-name]  (str/split full-plan-name #"::")
+        url (str "https://localhost:8140/puppet/v3/plans/" module-name "/" plan-name
+                 (if env-name (str "?environment=" env-name)))]
+    (try
+      (http-client/get url request-as-text-with-ssl)
+      (catch Exception e
+        (throw (Exception. "plan info http get failed" e))))))
+
+(defn parse-response
+  [response]
+  (-> response :body json/parse-string))
+
+(defn sort-plans
+  [plans]
+  (sort-by #(get % "name") plans))
+
+(def puppet-config
+  (-> (bootstrap/load-dev-config-with-overrides {:jruby-puppet
+                                                 {:gem-path [(ks/absolute-path jruby-testutils/gem-path)]
+                                                  :max-active-instances 1}})
+      (ks/dissoc-in [:jruby-puppet
+                     :environment-class-cache-enabled])))
+
+(deftest ^:integration all-plans-with-env
+  (testing "full stack plans listing smoke test"
+    (bootstrap/with-puppetserver-running-with-config
+      app
+      puppet-config
+      (do
+        (testutils/write-plans-files "apache" "announce" "return 'Hi!'")
+        (testutils/write-plans-files "graphite" "install" "return 'Wheeeee'")
+        (let [expected-response '({"name" "apache::announce"
+                                   "environment" [{"name" "production"
+                                                   "code_id" nil}]}
+                                  {"name" "graphite::install"
+                                   "environment" [{"name" "production"
+                                                   "code_id" nil}]})
+              response (get-all-plans "production")]
```

Comment:
Should we test for correctly returned `environment-not-found` output when the environment doesn't exist?

Reply:
Given that's handled by common helpers and we do so below, I don't think it's necessary